tests/async/test_browsertype_connect.py

```diff
@@ -0,0 +1,143 @@
+    assert len(browser1.contexts) == 1
+
+    await browser1.close()
+    page2 = await browser2.new_page()
+    # original browser should still work
+    assert await page2.evaluate("7 * 6") == 42
+
+    await browser2.close()
+
+
+async def test_browser_type_connect_disconnected_event_should_be_emitted_when_browser_is_closed_or_server_is_closed(
+    browser_type: BrowserType, launch_server
+):
+    # Launch another server to not affect other tests.
+    remote = launch_server()
+
+    browser1 = await browser_type.connect(remote.ws_endpoint)
+    browser2 = await browser_type.connect(remote.ws_endpoint)
+
+    disconnected1 = []
+    disconnected2 = []
+    browser1.on("disconnected", lambda: disconnected1.append(True))
+    browser2.on("disconnected", lambda: disconnected2.append(True))
+
+    page2 = await browser2.new_page()
+
+    await browser1.close()
+    assert len(disconnected1) == 1
+    assert len(disconnected2) == 0
+
+    remote.kill()
+    assert len(disconnected1) == 1
+
+    with pytest.raises(Error):
+        # Tickle connection so that it gets a chance to dispatch disconnect event.
+        await page2.title()
+    assert len(disconnected2) == 1
+
+
+async def test_browser_type_disconnected_event_should_have_browser_as_argument(
+    browser_type: BrowserType, launch_server
+):
+    remote_server = launch_server()
+    browser = await browser_type.connect(remote_server.ws_endpoint)
+    event_payloads = []
+    browser.on("disconnected", lambda b: event_payloads.append(b))
+    await browser.close()
+    assert event_payloads[0] == browser
+
+
+async def test_browser_type_connect_set_browser_connected_state(
+    browser_type: BrowserType, launch_server
+):
+    remote_server = launch_server()
+    browser = await browser_type.connect(remote_server.ws_endpoint)
+    assert browser.is_connected()
+    await browser.close()
+    assert browser.is_connected() is False
+
+
+async def test_browser_type_connect_should_throw_when_used_after_is_connected_returns_false(
+    browser_type: BrowserType, launch_server
+):
+    remote_server = launch_server()
+    browser = await browser_type.connect(remote_server.ws_endpoint)
+    page = await browser.new_page()
+
+    remote_server.kill()
+
+    with pytest.raises(Error) as exc_info:
+        await page.evaluate("1 + 1")
+    assert "Playwright connection closed" == exc_info.value.message
+    assert browser.is_connected() is False
+
+
+async def test_browser_type_connect_should_reject_navigation_when_browser_closes(
+    server: Server, browser_type: BrowserType, launch_server
+):
+    remote_server = launch_server()
+    browser = await browser_type.connect(remote_server.ws_endpoint)
+    page = await browser.new_page()
+    await browser.close()
+
+    with pytest.raises(Error) as exc_info:
+        await page.goto(server.PREFIX + "/one-style.html")
+    assert "Playwright connection closed" in exc_info.value.message
+
+    with pytest.raises(Error) as exc_info:
```

Comment:
Remove the last assertions as it is ducplicated

Reply:
intended to ensure it works multiple times.